clm/clm_datasets.py

```diff
@@ -169,4 +227,4 @@ def mask_pad(examples):
             [lm_datasets["validation"], pad_validation_dataset]
         )
 
-    return lm_datasets
+    return lm_datasets.with_format("torch")
```

Comment:
I got type error like
```
Traceback (most recent call last):
 ...
  File "/app/MoE_study/clm/run_clm.py", line 97, in main
    datasets = process_datasets(raw_datasets, tokenizer, config)
  File "/app/MoE_study/clm/clm_datasets.py", line 230, in process_datasets
    return lm_datasets.with_format("torch")
AttributeError: 'dict' object has no attribute 'with_format'
```
And I can wrap the dictionary `lm_datasets` with hf's `DatasetDict` api and solve the type error. However it doesn't work well with `IterableDataset` or stream mode.
```
from datasets import DatasetDict
DatasetDict(lm_datasets).with_format("torch")
```

Got the following error from IterableDataset in stream mode:
```
    return DatasetDict(lm_datasets).with_format("torch")
  File "/usr/local/lib/python3.10/site-packages/datasets/dataset_dict.py", line 725, in with_format
    dataset.set_format(type=type, columns=columns, output_all_columns=output_all_columns, **format_kwargs)
  File "/usr/local/lib/python3.10/site-packages/datasets/dataset_dict.py", line 596, in set_format
    self._check_values_type()
  File "/usr/local/lib/python3.10/site-packages/datasets/dataset_dict.py", line 49, in _check_values_type
    raise TypeError(f"Values in `DatasetDict` should be of type `Dataset` but got type '{type(dataset)}'")
TypeError: Values in `DatasetDict` should be of type `Dataset` but got type '<class 'datasets.iterable_dataset.IterableDataset'>'
```